src/main/java/com/ufcg/es5/BackendComplexoEsportivoUFCG/application/register_confirmation_code/service/SignUpConfirmationCodeServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+package com.ufcg.es5.BackendComplexoEsportivoUFCG.application.register_confirmation_code.service;
+
+import com.ufcg.es5.BackendComplexoEsportivoUFCG.application.event.SavedEvent;
+import com.ufcg.es5.BackendComplexoEsportivoUFCG.application.register_confirmation_code.repository.SignUpConfirmationCodeRepository;
+import com.ufcg.es5.BackendComplexoEsportivoUFCG.application.sace_user.service.SaceUserService;
+import com.ufcg.es5.BackendComplexoEsportivoUFCG.dto.sign_up_confirmation_code.SignUpConfirmationCodeUserIdConfirmationCodeDto;
+import com.ufcg.es5.BackendComplexoEsportivoUFCG.entity.SaceUser;
+import com.ufcg.es5.BackendComplexoEsportivoUFCG.entity.SignUpConfirmationCode;
+import com.ufcg.es5.BackendComplexoEsportivoUFCG.exception.common.SaceConflictException;
+import com.ufcg.es5.BackendComplexoEsportivoUFCG.exception.common.SaceResourceNotFoundException;
+import com.ufcg.es5.BackendComplexoEsportivoUFCG.exception.constants.sace_user.SaceUserExceptionMessages;
+import com.ufcg.es5.BackendComplexoEsportivoUFCG.util.security.RandomStringGenerator;
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.Collection;
+
+@Service
+public class SignUpConfirmationCodeServiceImpl implements SignUpConfirmationCodeService{
+
+
+    @Autowired
+    private SignUpConfirmationCodeRepository repository;
+
+    @Autowired
+    private SaceUserService userService;
+
+    @Autowired
+    private ApplicationEventPublisher eventPublisher;
+
+    @Override
+    public JpaRepository<SignUpConfirmationCode, Long> getRepository() {
+        return repository;
+    }
+
+    @Override
+    @Transactional
+    public void generateAndSend(Long userId) {
```

Comment:
Separe em duas funções, senão for possível dê um nome melhor que englobe as duas funcionalidades.